server/pairing/client.go

```diff
@@ -38,31 +39,53 @@ type BaseClient struct {
 	challengeTaker *ChallengeTaker
 }
 
-func findServerCert(c *ConnectionParams) (*url.URL, *x509.Certificate, error) {
-	netIps, err := server.FindReachableAddressesForPairingClient(c.netIPs)
-	if err != nil {
-		return nil, nil, err
-	}
+func findServerCert(c *ConnectionParams, reachableIPs []net.IP) (*url.URL, *x509.Certificate, error) {
 	var baseAddress *url.URL
 	var serverCert *x509.Certificate
-	var certErrs error
-	for _, ip := range netIps {
-		u := c.BuildURL(ip)
-
-		serverCert, err = getServerCert(u)
-		if err != nil {
-			var certErr string
-			if certErrs != nil {
-				certErr = certErrs.Error()
+	var once sync.Once
+	errCh := make(chan struct {
+		ip  net.IP
+		err error
+	}, len(reachableIPs))
+	certCh := make(chan *x509.Certificate, len(reachableIPs))
+	urlCh := make(chan *url.URL, len(reachableIPs))
```

Comment:
Maybe use a `struct`  for success channel too?
```suggestion
	certCh := make(chan struct {
	    cert *x509.Certificate,
	    url *url.URL
	}, len(reachableIPs))
```

And actually it seems that it can also be merged with `done` channel?